lhotse/recipes/wenet_speech.py

```diff
@@ -0,0 +1,141 @@
+    validate_recordings_and_supervisions,
+)
+from lhotse.audio import AudioSource, Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike
+
+WETNET_SPEECH_PARTS = ("L", "M", "S", "DEV", "TEST_NET", "TEST_MEETING")
+
+
+def prepare_wenet_speech(
+    corpus_dir: Pathlike,
+    dataset_parts: Union[str, Sequence[str]] = "all",
+    output_dir: Optional[Pathlike] = None,
+    num_jobs: int = 1,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Returns the manifests which consist of the Recordings and Supervisions
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param dataset_parts: Which parts of dataset to prepare, all for all the
+                          parts.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :num_jobs Number of workers to extract manifests.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with
+             the keys 'recordings' and 'supervisions'.
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    subsets = WETNET_SPEECH_PARTS if "all" in dataset_parts else dataset_parts
+
+    manifests = defaultdict(dict)
+    for sub in subsets:
+        if sub not in WETNET_SPEECH_PARTS:
+            raise ValueError(f"No such part of dataset in WenetSpeech : {sub}")
+        manifests[sub] = {"recordings": [], "supervisions": []}
+
+    raw_manifests_path = corpus_dir / "WenetSpeech.json"
+    assert raw_manifests_path.is_file(), f"No such file : {raw_manifests_path}"
+    logging.info(f"Loading raw manifests from : {raw_manifests_path}")
+    raw_manifests = json.load(open(raw_manifests_path, "r", encoding="utf8"))
+
+    with ProcessPoolExecutor(num_jobs) as ex:
+        for recording, segments in tqdm(
+            ex.map(
+                parse_utterance,
+                raw_manifests["audios"],
+                repeat(corpus_dir),
+                repeat(subsets),
+            ),
+            desc="Processing WenetSpeech JSON entries",
+        ):
+            for part in segments:
+                manifests[part]["recordings"].append(recording)
+                manifests[part]["supervisions"].extend(segments[part])
+
+    for sub in subsets:
+        recordings, supervisions = fix_manifests(
+            recordings=RecordingSet.from_recordings(manifests[sub]["recordings"]),
+            supervisions=SupervisionSet.from_segments(manifests[sub]["supervisions"]),
+        )
+        validate_recordings_and_supervisions(
+            recordings=recordings, supervisions=supervisions
+        )
+
+        if output_dir is not None:
+            supervisions.to_json(output_dir / f"supervisions_{sub}.json")
+            recordings.to_json(output_dir / f"recordings_{sub}.json")
```

Comment:
I suggest writing these out to JSONL format (one JSON per line) as it enables to read the Lhotse manifest lazily. I expect these are also possibly quite large, so you could also gzip them. It'd look like:

```suggestion
            recordings.to_file(output_dir / f"recordings_{sub}.jsonl.gz")
```

(the same for supervision manifest)